### ISSUE TITLE
Bug: CPU loading fails when device=torch.device("cpu")

### ISSUE THREAD

**Issue by hmirheydari** [resemble-ai/chatterbox#533]
Environment:
- Windows 11
- Python 3.11
- CPU-only machine

Code:

from chatterbox.mtl_tts import ChatterboxMultilingualTTS
import torch

model = ChatterboxMultilingualTTS.from_pretrained(
    device=torch.device("cpu")
)

Result:

RuntimeError:
Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False

Cause:

In D:\chatterbox\Lib\site-packages\chatterbox\mtl_tts.py
In from_local():

if device in ["cpu", "mps"]:
    map_location = torch.device("cpu")
else:
    map_location = None

When device is torch.device("cpu"), the condition evaluates to False.

**Fix**
Instead of
`if device in ["cpu", "mps"]:
    map_location = torch.device("cpu")
else:
    map_location = None`

replace:

`if str(device) in ["cpu", "mps"]:
    map_location = torch.device("cpu")
else:
    map_location = None`

**Comment by li-lizhe** [comment]
Addressed in #554 — the root cause is that `torch.device("cpu")` does not compare equal to the string `"cpu"`, so `device in ["cpu", "mps"]` is `False` and `from_local()` fell through to `map_location=None`.

The fix adds a shared `load_map_location()` helper that normalizes the device via `torch.device(device)` and returns `map_location=None` only for CUDA. Every non-CUDA accelerator (CPU/MPS/NPU/XPU) now loads onto CPU first. The same `torch.device` vs `str` comparison bug in the MPS availability check is fixed too.

Verified on an Ascend 910B NPU (`load_map_location("npu")` / `load_map_location(torch.device("npu"))` both resolve to CPU).
